src/main/java/com/puppycrawl/tools/checkstyle/gui/CodeSelectorPModel.java

```diff
@@ -46,7 +47,18 @@
      * @param lines2position list to map lines.
      */
     public CodeSelectorPModel(DetailAST ast, List<Integer> lines2position) {
-        this.ast = ast;
+        node = ast;
+        final List<Integer> copy = new ArrayList<>(lines2position);
+        this.lines2position = Collections.unmodifiableList(copy);
```

Comment:
This was before my editing:
https://github.com/checkstyle/checkstyle/pull/3554/files/d27161d4491ce0707d93c9d1426dc920ede1c7b6#diff-cf9b6157630180a13db71355b2241bdfL50